modules/install/optional_write_files.sh

```diff
@@ -7,7 +7,7 @@ files=$1
 
 if [ -n "$files" ]
 then
-  file_array=($(echo "$files" | tr ' ' '\n'))
+  read -r -a file_array <<< "$files"
```

Comment:
I remember writing this a while back and getting shellchecker being upset with me for not doing it this way, but in certain OSes this didn't work. Have you confirmed this works at least in a critical subset of our supported OSes? (SLE Micro, SLES, RHEL, Ubuntu). Don't need to run the full automation, just the script itself should be fine.

Same comment with other shell file modifications, but I assume if it works in one then it'll work in the others.

Reply:
checking

Reply:
Do we want to have shellcheck results run on the repo itself? I know in my own scripts I have needed to add some ignore sections, because the shellcheck rule was breaking my working across all OS under test functionality.

Reply:
i think we need to evaluate each case/warning, sometimes really makes sense to add ignore , in other cases really makes sense to adapt or to comply

Reply:
but regarding to this specific one, i am rolling back the changes, since it actually does not work well on some Os.s

Reply:
Is the change reverted?

Reply:
I think still need to rollback these changes